cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/BaseStorage.java

```diff
@@ -76,4 +83,34 @@ public String allocateTableLocation(
         .normalize()
         .toString();
   }
+
+  /**
+   * Deallocates/deletes Table Storage location Walk through the directory structure, sort the paths
+   * in reverse order to ensure files are deleted before directories, and deletes each path.
+   *
+   * @param location the base location of the table
+   * @param tableCreator the creator of the table
+   * @throws IOException
+   */
+  @Override
+  public void deallocateTableLocation(String location, String tableCreator) throws IOException {
+    List<IOException> exceptions = new ArrayList<>();
```

Comment:
let's leave this as a no-op for `BaseStorage` and add some log for example: "deallocateTableLocation action not defined, skipping deallocation for Storage: XYZ"

Files.walk code will likely not work for storages outside of local system like gcs, or s3 or abs.